cluster/get-kube-binaries.sh

```diff
@@ -222,22 +198,47 @@ if [[ -z "${KUBERNETES_SKIP_CONFIRM-}" ]]; then
   fi
 fi
 
-if "${DOWNLOAD_SERVER_TAR}"; then
-  download_tarball "${KUBE_ROOT}/server" "${SERVER_TAR}"
-fi
+download_tarball "${KUBE_ROOT}/server" "${SERVER_TAR}"
 
 if "${DOWNLOAD_NODE_TAR}"; then
   download_tarball "${KUBE_ROOT}/node" "${NODE_TAR}"
 fi
 
-if "${DOWNLOAD_CLIENT_TAR}"; then
-  download_tarball "${KUBE_ROOT}/client" "${CLIENT_TAR}"
-  extract_arch_tarball "${KUBE_ROOT}/client/${CLIENT_TAR}" "${CLIENT_PLATFORM}" "${CLIENT_ARCH}"
-fi
+download_tarball "${KUBE_ROOT}/client" "${CLIENT_TAR}"
+extract_arch_tarball "${KUBE_ROOT}/client/${CLIENT_TAR}" "${CLIENT_PLATFORM}" "${CLIENT_ARCH}"
+ln -s "${KUBE_ROOT}/platforms/${CLIENT_PLATFORM}/${CLIENT_ARCH}" "${KUBE_ROOT}/client/bin"
+echo "Add '${KUBE_ROOT}/client/bin' to your PATH to use newly-installed binaries."
 
 if "${DOWNLOAD_TESTS_TAR}"; then
-  download_tarball "${KUBE_ROOT}/test" "${TESTS_TAR}"
-  echo "Extracting ${TESTS_TAR} into ${KUBE_ROOT}"
-  # Strip leading "kubernetes/"
-  tar -xzf "${KUBE_ROOT}/test/${TESTS_TAR}" --strip-components 1 -C "${KUBE_ROOT}"
+  TESTS_PORTABLE_TAR="kubernetes-test-portable.tar.gz"
+  download_tarball "${KUBE_ROOT}/test" "${TESTS_PORTABLE_TAR}" || true
+  if [[ -f "${KUBE_ROOT}/test/${TESTS_PORTABLE_TAR}" ]]; then
+    echo "Extracting ${TESTS_PORTABLE_TAR} into ${KUBE_ROOT}"
+    # Strip leading "kubernetes/"
+    tar -xzf "${KUBE_ROOT}/test/${TESTS_PORTABLE_TAR}" --strip-components 1 -C "${KUBE_ROOT}"
+
+    # Next, download platform-specific test tarballs for all relevant platforms
+    TEST_PLATFORM_TUPLES=(
+      "${CLIENT_PLATFORM}/${CLIENT_ARCH}"
+      "${SERVER_PLATFORM}/${SERVER_ARCH}"
+      )
+    if [[ -n "${NODE_PLATFORM:-}" && -n "${NODE_ARCH:-}" ]]; then
+      TEST_PLATFORM_TUPLES+=("${NODE_PLATFORM}/${NODE_ARCH}")
+    fi
+    # Loop over only the unique tuples
+    for TUPLE in $(printf "%s\n" "${TEST_PLATFORM_TUPLES[@]}" | sort -u); do
+        OS=$(echo "${TUPLE}" | cut -d/ -f1)
+        ARCH=$(echo "${TUPLE}" | cut -d/ -f2)
```

Comment:
who says bash can't do sets of tuples

Reply:
I could probably use `IFS="/" read` here instead, but I felt this was more readable.

Reply:
who says indeed? :P